### PR TITLE
CMR-5061: Added support of AccessConstraints in UMM-G.

### PR DESCRIPTION
**Pull request by ygliuvt**

### CODE REVIEW

File: umm-spec-lib/src/cmr/umm_spec/umm_g/granule.clj [hunk 1]
```diff
@@ -108,6 +108,7 @@
                                              (str ending-date-time))}}))
      :Platforms (platform/PlatformRefs->umm-g-platforms platform-refs)
      :CloudCover cloud-cover
+     :AccessConstraints (when access-value {:Value access-value})
```
Comment (daniel-zamora) [inline umm-spec-lib/src/cmr/umm_spec/umm_g/granule.clj]: I think this when is redundant.

Comment (daniel-zamora) [reply umm-spec-lib/src/cmr/umm_spec/umm_g/granule.clj]: I don't think this works.

Comment (daniel-zamora) [reply umm-spec-lib/src/cmr/umm_spec/umm_g/granule.clj]: ```:access-value 42``` Not a map.

Comment (ygliuvt) [reply umm-spec-lib/src/cmr/umm_spec/umm_g/granule.clj]: This is generating UMM-G `AccessConstraints` which is a map, not umm-lib `access-value`. We don't want to make an empty map when there is no `access-value`, thus the `when`.

**Pull request merged**